app/src/main/java/com/google/samples/apps/sunflower/PlantDetailFragment.kt

```diff
@@ -56,11 +57,41 @@ class PlantDetailFragment : Fragment() {
         val binding = DataBindingUtil.inflate<FragmentPlantDetailBinding>(
                 inflater, R.layout.fragment_plant_detail, container, false).apply {
             viewModel = plantDetailViewModel
-            lifecycleOwner = this@PlantDetailFragment
+            lifecycleOwner = viewLifecycleOwner
             fab.setOnClickListener { view ->
+                hideAppBarFab(fab)
                 plantDetailViewModel.addPlantToGarden()
                 Snackbar.make(view, R.string.added_plant_to_garden, Snackbar.LENGTH_LONG).show()
             }
+
+            // scroll change listener begins at Y = 0 when image is fully collapsed
+            plantDetailScrollview.setOnScrollChangeListener(NestedScrollView.OnScrollChangeListener { _, _, scrollY, _, _ ->
+                // user scrolled past image to height of toolbar
+                val showToolbar = scrollY > toolbar.height
+
+                // set title programatically so not shown by default
+                toolbarLayout.title = viewModel?.plant?.value?.name
```

Comment:
Should this be in the scroll listener?

Reply:
Moved setting the title into the layout.  Also added logic to only update the appbar and toolbar attributes when the scroll threshold is crossed, instead of continuously during scrolling